src/Core/ViewHelper/ViewHelperVariableContainer.php

```diff
@@ -29,6 +30,69 @@ class ViewHelperVariableContainer
      */
     protected $view;
 
+    /**
+     * @var VariableProviderInterface[]
+     */
+    protected $delegates = [];
+
+    /**
+     * Push a new delegate variable container to a stack.
+     *
+     * If a ViewHelper requires a storage to collect variables which, for
+     * example, are filled by evaluating the child (tag content) closure,
+     * this method can be used to add a special delegate variable container
+     * stored in a stack. Once the variables you need to collect have been
+     * collected, calling `popDelegateVariableProvider` removes the delegate
+     * from the stack.
+     *
+     * The point of a stack is to avoid resetting a storage every time a
+     * ViewHelper is rendered. In the case of `f:render` it means one storage
+     * is created and filled for every one call to the ViewHelper.
+     *
+     * It is VITAL that you also "pop" any delegate you push to this stack!
+     *
+     * @param VariableProviderInterface $variableProvider
+     */
+    public function pushDelegateVariableProvider(VariableProviderInterface $variableProvider)
+    {
+        $this->delegates[] = $variableProvider;
+    }
+
+    /**
+     * Get the topmost delegate variable container that was previously pushed
+     * onto the stack by pushDelegateVariableContainer(). This method returns
+     * a reference to the storage that was last added to the stack without
+     * removing the variable provider from the stack.
+     *
+     * Is used in ViewHelpers that assign variables in variable providers in
+     * the stack - as a means to get the variable storage used by the "closest
+     * parent", e.g. when called in `f:argument` used inside `f:render`, will
+     * read the delegate variable provider inserted by that parent `f:render`.
+     *
+     * @return VariableProviderInterface|null
+     */
+    public function getTopmostDelegateVariableProvider()
+    {
+        return end($this->delegates) ?: null;
```

Comment:
Would it make sense to fall back to `$this`? This way any calling code does not have to care about what and if something was pushed to the stack.

Reply:
`$this` is a ViewHelperVariableContainer and this has a different signature for `add` and will throw exceptions. Automatically adding one to the stack isn't sensible either since and empty stack means you used the VH outside of proper context.